sql/codegen_xgboost.go

```diff
@@ -0,0 +1,606 @@
+	"colsample_bytree":     fp32Partial("colsample_bytree", func(r *xgboostFiller) *float32 { return &(r.ColSampleByTree) }),
+	"colsample_bylevel":    fp32Partial("colsample_bylevel", func(r *xgboostFiller) *float32 { return &(r.ColSampleByLevel) }),
+	"max_bin":              uIntPartial("max_bin", func(r *xgboostFiller) *uint { return &(r.MaxBin) }),
+	"convergence_criteria": strPartial("convergence_criteria", func(r *xgboostFiller) *string { return &(r.ConvergenceCriteria) }),
+	"verbosity":            uIntPartial("verbosity", func(r *xgboostFiller) *uint { return &(r.Verbosity) }),
+	// xgboost train controllers
+	"num_round":  uIntPartial("num_round", func(r *xgboostFiller) *uint { return &(r.NumRound) }),
+	"auto_train": boolPartial("auto_train", func(r *xgboostFiller) *bool { return &(r.AutoTrain) }),
+	// xgboost output columns (for prediction)
+	"append_columns":  sListPartial("append_columns", func(r *xgboostFiller) *[]string { return &(r.AppendColumns) }),
+	"result_column":   strPartial("result_column", func(r *xgboostFiller) *string { return &(r.ResultColumn) }),
+	"prob_column":     strPartial("prob_column", func(r *xgboostFiller) *string { return &(r.ProbColumn) }),
+	"detail_column":   strPartial("detail_column", func(r *xgboostFiller) *string { return &(r.DetailColumn) }),
+	"encoding_column": strPartial("encoding_column", func(r *xgboostFiller) *string { return &(r.EncodingColumn) }),
+	// Label, Group, Weight and xgFeatureFields are parsed from columnClause
+}
+
+func xgParseAttr(pr *extendedSelect, r *xgboostFiller) error {
+	// parse pr.attrs to map[string][]string
+	attrs := make(map[string][]string)
+	for k, exp := range pr.attrs {
+		strExp := exp.String()
+		if strings.HasPrefix(strExp, "[") && strings.HasSuffix(strExp, "]") {
+			attrs[k] = exp.cdr()
+		} else {
+			attrs[k] = []string{strExp}
+		}
+		for i, s := range attrs[k] {
+			if s[0] == 34 && s[len(s)-1] == 34 {
+				s = s[1 : len(s)-1]
+				attrs[k][i] = s
+			}
+		}
+	}
+
+	// fill xgboostFiller with attrs
+	for k := range attrs {
+		if setter, ok := xgbAttrSetterMap[k]; ok {
+			if e := setter(&attrs, r); e != nil {
+				return xgParseAttrError(e)
+			}
+		}
+	}
+
+	// remaining elements in xgbAttrs are unsolved ones, so we throw exception if any elements remaining.
+	if len(attrs) > 0 {
+		for k, v := range attrs {
+			log.Errorf("unsolved xgboost attr: %s = %s", k, v)
+		}
+		return fmt.Errorf("found unsolved xgboost attributes")
+	}
+
+	return nil
+}
+
+/* parse feature column, which owned by default column target("feature_columns:), from AST(pr.columns)
+
+   	For now, two schemas are supported:
+		1. sparse-kv
+			schema: COLUMN SPARSE([feature_column], [1-dim shape], [single char delimiter])
+			data example: COLUMN SPARSE("0:1.5 1:100.1f 11:-1.2", [20], " ")
+		2. tf feature columns
+			 roughly same as TFEstimator, except output shape of feaColumns are required to be 1-dim.
+*/
+func parseFeatureColumns(columns *exprlist, r *xgboostFiller) error {
```

Comment:
Not sure why the CI's linter check didn't report this style check that, comments should start with the function name, see: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

Reply:
@typhoonzero maybe linter only check exposed functions?